olive/platform_sdk/qualcomm/utils/data_loader.py

```diff
@@ -17,27 +16,29 @@
 logger = logging.getLogger(__name__)
 
 
-class FileListDataLoader(ABC):
-    """Abstraction for logical "FileListDataLoader", it contains data path and related metadata."""
+class FileListDataLoader:
+    """Data loader for use with Qualcomm SDKs that loads data from a file list."""
 
-    def __init__(self, config: dict, batch_size: int = None):
+    def __init__(self, load_data_func: Callable, batch_size: int = None, **kwargs):
         # TODO(anyone): try to add file_chunk_size to distinguish the concept of batch_size and file_chunk_size
-        """:param config: data loader specific config."""
-        self.config = config
+        """Initialize FileListDataLoader.
+
+        :param load_data_func: function to load data. Must take tmp_dir_name and **kwargs, and return a tuple
+            of (data_dir, input_list, annotation).
+        :param batch_size: number of inputs per chunked input list file for batch processing. If None, all inputs are in
+            a single input list file.
+        :param kwargs: additional arguments to pass to load_data_func.
+        """
         self.batch_size = batch_size
-        self.tmp_dir = None
-        self.data_dir, self.input_list, self.annotation = self.load_data()
+        self.tmp_dir = tempfile.TemporaryDirectory(prefix="olive_tmp_")  # pylint: disable=consider-using-with
+        self.data_dir, self.input_list, self.annotation = load_data_func(self.tmp_dir.name, **kwargs)
```

Comment:
could we use dedicated creation function or class method to construct the dataloader? 
in which, we could 
1. initiate the object
2. call load_data
3. call prepare_data